lock_api/Cargo.toml

```diff
@@ -13,5 +13,14 @@ edition = "2018"
 scopeguard = { version = "1.0", default-features = false }
 owning_ref = { version = "0.4", optional = true }
 
+# Optional dependency for supporting serde.  Use the feature "enable_serde" to
+# get access to this.
+serde = {version = "1.0.90", features = ["derive", "rc", "std"], optional = true}
```

Comment:
Optional dependencies are already treated as feature names. So you can just use `cfg(feature = "serde)` and get rid of the "enable_serde" feature.

Reply:
Also, why are the "derive", "rc" and "std" features enabled? I don't think any of these are needed. In fact you should be using `default-features = false` so that no_std support for serde is enabled.

Reply:
> Optional dependencies are already treated as feature names. So you can just use `cfg(feature = "serde)` and get rid of the "enable_serde" feature.

I didn't know that; I'm still learning rust, so thank you for teaching me something new!

> Also, why are the "derive", "rc" and "std" features enabled? I don't think any of these are needed. In fact you should be using `default-features = false` so that no_std support for serde is enabled.

I enabled them because they mirrored how I was using them in my own project.  As long as all those features can be enabled in a different project easily, then there's no need to enable them.  Would the normal method to enable them in a project that uses `parking_lot` be something like:
```toml
serde = {features=["parking_lot/serde/rc"]}
```
or would it be
```toml
serde = {features=["serde/rc"]}
```
?

As long as the method to enable the features is well-documented so that third party users can enable them as needed, then I think removing the requirements is a good idea.

Reply:
If your own crate needs to use serde, then it will itself depend on serde and enable the features that it needs. Cargo will combine the serde features from all dependencies and enable all the ones that are needed.

In the case of lock_api, we only want to enable the bare minimum of features, so that we are maximally compatible. This is why we disable even the default serde features in lock_api so that we don't enable the "std" feature and can be compatible with no_std projects.

Reply:
> Cargo will combine the serde features from all dependencies and enable all the ones that are needed.

I didn't know that; thanks for letting me know!  Now changed.